src/newtondynamics_physics/ospnewton.h

```diff
@@ -38,30 +46,35 @@ struct DeleterNewtonBody
     { NewtonDestroyBody(pCollision); }
 };
 
-using ACompNwtBody_t = std::unique_ptr<NewtonBody const, DeleterNewtonBody>;
+using NwtBodyPtr_t = std::unique_ptr<NewtonBody, DeleterNewtonBody>;
 
 struct DeleterNewtonCollision
```

Comment:
You might consider making a utility class like utils/pointer_to_function.h
```
template<auto FUNCTION>
struct PointerToFunctionAdapter
{
    static_assert(nullptr != FUNCTION)
    decltype(auto) operator()(auto && ... args)
    {
        return std::invoke(FUNCTION, std::forward<decltype(args)>(args)...);
    }
};
```

This lets you define the struct used for your unique_ptr once, and then save on having to define a new struct.

Reply:
I like this idea. No need to be too generalized imo, this seems to work too:
```cpp
template<auto FUNC_T>
struct NwtDeleter
{
    void operator() (auto const* pNwtType)
    {
        FUNC_T(pNwtType);
    }
};
```